src/ProtectedRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { useContext } from 'react';
+import { Route, Redirect, RouteProps } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { AuthContext } from './context/AuthProvider';
+
+const ProtectedRoute: React.FC<RouteProps> = ({ exact, path, component }) => {
+  const user = useContext(AuthContext);
+
+  if (user === undefined) {
+    return <p>Loading...</p>;
+  }
+
+  if (user) {
+    return <Route exact={exact} path={path} component={component} />;
+  }
+
+  return <Redirect to="/" />;
+};
+
+ProtectedRoute.defaultProps = {
+  exact: true
```

Comment:
I get your point right now. We always want to have exact as true and only set it to false if it's really necessarry